app/src/main/java/com/hedvig/app/feature/embark/util/VariableExtractor.kt

```diff
@@ -0,0 +1,123 @@
+package com.hedvig.app.feature.embark.util
+
+import com.hedvig.android.owldroid.fragment.GraphQLVariablesFragment
+import com.hedvig.android.owldroid.type.EmbarkAPIGraphQLSingleVariableCasting
+import com.hedvig.android.owldroid.type.EmbarkAPIGraphQLVariableGeneratedType
+import com.hedvig.app.feature.embark.ValueStore
+import com.hedvig.app.util.plus
+import com.hedvig.app.util.toJsonObject
+import java.util.UUID
+import org.json.JSONObject
+
+object VariableExtractor {
+
+    fun extractVariables(variables: List<GraphQLVariablesFragment>, valueStore: ValueStore): JSONObject {
+        var regularVariables = variables.mapNotNull {
+            extractRegularVariables(it, valueStore)
+        }.toJsonObject()
+
+        val multiActionVariables = variables.mapNotNull {
+            extractMultiActionVariables(it, valueStore)
+        }
+
+        multiActionVariables.forEach {
+            regularVariables = regularVariables.plus(it)
+        }
+        return regularVariables
+    }
+
+    private fun extractRegularVariables(
```

Comment:
Could we split this into two methods, `extractSingleVariable`/`extractGeneratedVariable` which as input take `GraphQLVariablesFragment.AsGraphQLSingleVariable`/`GraphQLVariablesFragment.AsGeneratedVariable`? That would remove the need for having a possible null return for this/these functions.